docs/modules/chains/examples/smart_llm.ipynb

```diff
@@ -0,0 +1,258 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "9e9b7651",
+   "metadata": {},
+   "source": [
+    "# How to use a SmartLLMChain\n",
+    "\n",
+    "A SmartLLMChain is a form of self-critique chain that can help you if have particularly complex questions to answer. Instead of doing a single LLM pass, it instead performs these 3 steps:\n",
+    "1. Ideation: Pass the user prompt n times through the LLM to get n output proposals (called \"ideas\"), where n is a parameter you can set \n",
+    "2. Critique: The LLM critiques all ideas to find possible flaws and picks the best one \n",
+    "3. Resolve: The LLM tries to improve upon the best idea (as chosen in the critique step) and outputs it. This is then the final output.\n",
+    "\n",
+    "SmartLLMChains are based on the SmartGPT workflow proposed in https://youtu.be/wVzuvf9D9BU.\n",
+    "\n",
+    "Note that SmartLLMChains\n",
+    "- use more LLM passes (ie n+2 instead of just 1)\n",
+    "- only work then the underlying LLM has the capability for reflection, whicher smaller models often don't\n",
+    "- only work with underlying models that return exactly 1 output, not multiple\n",
+    "\n",
+    "This notebook demonstrates how to use a SmartLLMChain."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "714dede0",
+   "metadata": {},
+   "source": [
+    "##### Same LLM for all steps"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "d3f7fb22",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.environ[\"OPENAI_API_KEY\"] = \"...\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "10e5ece6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.prompts import PromptTemplate\n",
+    "from langchain.chat_models import ChatOpenAI\n",
+    "from langchain.chains import SmartLLMChain"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "8049cecd",
+   "metadata": {},
+   "source": [
+    "We first create an LLM and prompt template"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "811ea8e1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hard_question = \"I have a 12 liter jug and a 6 liter jug. I want to measure 6 liters. How do I do it?\"\n",
```

Comment:
Sorry, I didn't get it. 
To measure 6 liter you have to fill in the 6 liter jug. That's all.
The answer definitely didn't provide a good answer.
Probably, the question can be like the puzzle for the "Die hard" move:
Given a 5-gallon jug and a 3-gallon jug. Please, fill one with exactly 4 gallons of water.

Reply:
Hey, I disagree :) This is the example shown in the original SmartGPT video (https://youtu.be/wVzuvf9D9BU?t=384). The point is none of the original ideas work at all (even with GPT4!), only with the SmartGPT workflow do we get a correct answer.

But I agree this is confusing without context. I added more explanation.